library/Pipeline/src/Juvix/Pipeline.hs

```diff
@@ -47,9 +65,58 @@ class HasBackend b where
         Pipeline.toCore
           ("stdlib/Prelude.ju" : stdlibs b ++ [fp])
 
-  typecheck ::
+  typecheck :: Context.T Sexp.T Sexp.T Sexp.T -> Pipeline (ErasedAnn.AnnTermT (Ty b) (Val b))
+
+  typecheck' ::
+    ( Eq (Ty b),
+      Show (Ty b),
+      Eq (Val b),
+      Show (Err b),
+      Show (Val b),
+      CanApply (Ty b),
+      CanApply (TypedPrim (Ty b) (Val b)),
+      IR.HasWeak (Val b),
+      IR.HasSubstValue IR.NoExt (Ty b) (TypedPrim (Ty b) (Val b)) (Ty b),
+      IR.HasPatSubstTerm (OnlyExts.T IR.NoExt) (Ty b) (TypedPrim (Ty b) (Val b)) (Ty b),
+      Show (ApplyErrorExtra (Ty b)),
+      Show (ApplyErrorExtra (TypedPrim (Ty b) (Val b))),
+      Show (Arg (Ty b)),
+      Show (Arg (TypedPrim (Ty b) (Val b))),
+      IR.HasPatSubstTerm (OnlyExts.T IR.NoExt) (Ty b) (Val b) (Ty b),
+      IR.HasPatSubstTerm (OnlyExts.T IR.NoExt) (Ty b) (Val b) (Val b),
+      IR.HasPatSubstTerm (OnlyExts.T TypeChecker.T) (Ty b) (TypedPrim (Ty b) (Val b)) (Ty b)
+    ) =>
```

Comment:
Kind of an unwiedly signature, this shows we may need better instnaces in some regards? Cool how GHC can include it here